FlowCal/mef.py

```diff
@@ -308,38 +370,44 @@ def sc_fun(p,x):
             beads_model_str,
             beads_params_names)
 
-def plot_standard_curve(fl_channel,
+def plot_standard_curve(fl_au,
                         fl_mef,
                         beads_model,
                         std_crv,
+                        xscale='linear',
+                        yscale='linear',
                         xlim=None,
                         ylim=(1.,1e8)):
     """
     Plot a standard curve with fluorescence of calibration beads.
 
     Parameters
     ----------
-    fl_channel : array_like
-        Fluorescence of the calibration beads' subpopulations, in channel
-        numbers.
+    fl_au : array_like
+        Fluorescence of the calibration beads' subpopulations, in arbitrary
+        units.
     fl_mef : array_like
         Fluorescence of the calibration beads' subpopulations, in MEF
         units.
     beads_model : function
         Fluorescence model of the calibration beads.
     std_crv : function
-        The standard curve, mapping channel units to MEF units).
+        The standard curve, mapping arbitrary units to MEF units.
 
     Other parameters:
     -----------------
+    xscale : str, optional
+        Scale of the x axis, either ``linear`` or ``log``.
+    yscale : str, optional
+        Scale of the y axis, either ``linear`` or ``log``.
```

Comment:
> When I look at the standard curves, I try to quickly identify how many decades in a.u. are mapped to how many decades in MEF.

I admittedly *don't* try and identify how many decades in a.u. are mapped to how many decades are in MEF. I feel like you may be doing that due to your extensive exploration into the differences among amplifier (log, lin) and data (int, float) types.

Hadn't really thought about what I look for in a standard curve, but I guess it's (1) how closely the beads model fits the bead data points and (2) how significant the deviation on the low end is due to autofluorescence.

With those things in mind, this view definitely looks weird to me. I'm also confused why there aren't negative values on the top plot. Do you only show negative numbers if necessary? I'm struggling to identify where the linear-to-log transition occurs on the scales, too. These look just like log scales; is that just cause this was a quick attempt so the axes aren't right? or am I missing something?

> Another argument for keeping this plot in log scale is that the standard curve is a line in log scale, so this line should be easier to see if the plot itself is in log scale.

The standard curve should be a line on a *linear* scale now (RFI -> MEF) (almost... stupid `p[0]`... we even claimed in Note 3 that RFI and MEF are both supposed to be proportional to photons... boooo). Indeed, on a log-log plot, that should also still be a line, but I would push it even further and say that perhaps a *linear* scale would naively be most appropriate. In practice, we've discussed that it's more appropriate to go to a log space because are beads are logarithmically distributed in their fluorescence. I think because of *that* reason, you can justify a log plot view of the standard curve. If we were using linearly distributed beads, though, I think I would say a linear-linear plot would be most appropriate. This again all boils down to the fact that displaying this standard curve (which is *almost* a linear mapping from RFI -> MEF) and visualizing cell populations (which vary over many orders of magnitude but also have data near or below 0) are two fundamentally distinct visualization tasks, so it is not inconsistent to show one on log axes and the other on logicle axes.

> Finally, this function is not supposed to be accessed even by an API user, and instead used by get_transform_fxn. It is also not in the plot module, where I would agree it is a good idea to be consistent and make all functions work with logicle.

meh. I think these are weaker arguments. any function should still work as a stand-alone, in my opinion. And i think the plotting for the entire package should be consistent; nothing special about being in the `plot` module.


All this being said, I think I agree, though, that I wanna keep it as it was (log scale).